src/lxc/start.c

```diff
@@ -1414,8 +1414,16 @@ static int do_start(void *data)
 		#if HAVE_LIBCAP
 		if (lxc_proc_cap_is_set(CAP_SETGID, CAP_EFFECTIVE))
 		#endif
-			if (!lxc_setgroups(0, NULL))
-				goto out_warn_father;
+		{
+			if (handler->conf->init_groups.size > 0) {
+				if (!lxc_setgroups(handler->conf->init_groups.size,
+						   handler->conf->init_groups.list))
+					goto out_warn_father;
+			} else {
+				if (!lxc_setgroups(0, NULL))
+					goto out_warn_father;
+			}
+		}
```

Comment:
> No, similar to how you adapted `attach.c` to set groups on attach you should change `start.c where we call `lxc_setgroups()` as well.

I already did this in the first commit - is there something missing ?

I already did this in the first commit. Is there something missing ?